objcgen/objcgenerator.cs

```diff
@@ -151,15 +151,17 @@ protected override void Generate (Type t)
 			var native_name = GetTypeName (t);
 			headers.WriteLine ();
 			headers.WriteLine ($"// {t.AssemblyQualifiedName}");
-			headers.WriteLine ($"@interface {native_name} : {GetTypeName (t.BaseType)}");
```

Comment:
This is how it fails:

```
test.m:7:20: error: inconsistent number of instance variables specified
@public NSObject * _object;
                   ^
1 error generated.
```

Test case:
```objective-c
#include <Foundation/Foundation.h>

@interface X : NSObject
@end

@implementation X {
@public NSObject * _object;
}
@end

int main ()
{
	return 0;
}
```

built using:
```shell
clang test.m -framework Foundation -arch i386
```

See also: http://stackoverflow.com/a/15174088/183422